suite-common/message-system/src/messageSystemUtils.ts

```diff
@@ -327,3 +330,121 @@ export const resolveMessageContent = (localizedMessages: Localization, language:
 
     return localizedMessages[fallbackLanguage] ?? localizedMessages.en;
 };
+
+export const toMessageSystemOptions = <T extends string>(
+    values: readonly T[],
+): ReadonlyArray<{ label: string; value: T }> =>
+    values.map(value => ({
+        value,
+        label: value.replace(/^./, char => char.toUpperCase()),
+    }));
+
+export const getDefaultActionByCategory = (category: Category): Action => {
+    const defaultLocalization = {
+        en: '',
+        cs: '',
+        es: '',
+        de: '',
+        fr: '',
+        pt: '',
+    };
+
+    const baseMessage = {
+        id: uuidv4(),
+        priority: 100,
+        dismissible: true,
+        variant: 'info' as const,
+        category,
+        content: defaultLocalization,
+    };
+
+    const extraFields = (() => {
+        switch (category) {
+            case 'modal':
+                return {
+                    modal: {
+                        title: defaultLocalization,
+                        image: '',
+                    },
+                };
+            case 'context':
+                return {
+                    context: {
+                        domain: '',
+                    },
+                };
+            case 'feature':
+                return {
+                    feature: [
+                        {
+                            domain: '',
+                            flag: true,
+                        },
+                    ],
+                };
+            case 'banner':
+            default:
+                return {};
+        }
+    })();
+
+    return {
+        message: {
+            ...baseMessage,
+            ...extraFields,
+        },
+        conditions: [{}],
+    };
+};
+
+export const getDefaultConditionValue = (key: string) => {
```

Comment:
Declarative typing would be appreciated :pray: I believe this would be appropriate?
```suggestion
export const getDefaultConditionValue = (key: keyof Condition): Condition[keyof Condition] => {
```